src/charting/utils/Utils.ts

```diff
@@ -780,6 +780,10 @@ export namespace Utils {
         return remainder;
     }
 
+    export function getArrayItem(array: any[] | ObservableArray<any>, index: number) {
+        return array instanceof ObservableArray ? array.getItem(index) : array[index];
```

Comment:
2 things here.
First I thing that testing simply for `array.getItem` is faster than test for `instanceof`. We do that everywhere in N.
Also I am not sure we need to put that in Utils, I think putting it in `DataSet` is enough. the reasoning behind that `getEntryForIndex` will be faster as it does not have to call another function.
Also to prevent the need for the test on every call (which can be a lot if you have a lot of data) we can in `setValues` actually change `this.getEntryForIndex` to be either `(i)=>this.mValues[i]` or `(i)=>this.mValues.getItem(i)`
it s kind of dirty but so much faster

Reply:
About `instanceof`, I used it because tsc threw errors for `getItem` at certain cases.
Anyway, you can feel free to commit your own changes on this if you want or I can take a look at these in few hours. :)

Reply:
Sure let s merge this I work on it after